docroot/modules/humsci/hs_siteimprove/src/Plugin/Block/BrokenLinksBlock.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Drupal\hs_siteimprove\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Drupal\hs_siteimprove\SiteImprove;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a block displaying SiteImprove details.
+ *
+ * @Block(
+ *   id = "hs_siteimprove_broken_links",
+ *   admin_label = @Translation("SiteImprove Broken Links"),
+ *   category = @Translation("H&S Blocks")
+ * )
+ */
+class BrokenLinksBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The SiteImprove service.
+   *
+   * @var \Drupal\hs_siteimprove\SiteImprove
+   */
+  protected $siteImprove;
+
+  /**
+   * Constructs a new SiteImproveDetailsBlock instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\hs_siteimprove\SiteImprove $site_improve
+   *   The SiteImprove service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, SiteImprove $site_improve) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->siteImprove = $site_improve;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('hs_siteimprove.connector')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $pages = $this->siteImprove->getPagesWithBrokenLinks();
+    if (!$pages) {
+      return [
+        '#markup' => $this->t('No broken links found'),
+      ];
+    }
+
+    $build = [
+      'table' => [
+        '#type' => 'table',
+        '#header' => [
+          $this->t('Page'),
+          $this->t('Broken Links'),
+          $this->t('Actions'),
+        ],
+        '#rows' => $this->getRows($pages),
+        '#empty' => $this->t('No broken links found.'),
+        '#attributes' => [
+          'class' => ['siteimprove-broken-links-table'],
+        ],
+      ],
+      'full_report' => [
+        '#type' => 'link',
+        '#title' => $this->t('View Full Broken Links Report'),
+        '#url' => $this->getBrokenLinksReportUrl(),
+        '#attributes' => [
+          'class' => ['siteimprove-full-report-link'],
```

Comment:
We're standardizing the appearance of the "View more" links, let's change the class to re-use existing styles:
```suggestion
          'class' => ['button', 'button--primary'],
```